utopia-api/src/property-controls/property-controls.ts

```diff
@@ -21,71 +21,84 @@ export type BaseControlType =
   | 'vector3'
   | 'vector4'
 
-interface AbstractControlDescription<T extends ControlType> {
+export type CheckboxControlDescription = {
+  control: 'checkbox'
   label?: string
-  control: T
-  defaultValue?: unknown
-  visibleByDefault?: boolean
-}
-
-interface AbstractBaseControlDescription<T extends BaseControlType>
-  extends AbstractControlDescription<T> {}
-
-export interface CheckboxControlDescription extends AbstractBaseControlDescription<'checkbox'> {
   defaultValue?: boolean
+  visibleByDefault?: boolean
   disabledTitle?: string
   enabledTitle?: string
 }
 
-export interface ColorControlDescription extends AbstractBaseControlDescription<'color'> {
+export type ColorControlDescription = {
+  control: 'color'
+  label?: string
+  visibleByDefault?: boolean
   defaultValue?: string
 }
 
 export type AllowedEnumType = string | boolean | number | undefined | null
-export interface BasicControlOption<T> {
+export type BasicControlOption<T> = {
   value: T
   label: string
 }
 
 export type BasicControlOptions<T> = AllowedEnumType[] | BasicControlOption<T>[]
 
-export interface PopUpListControlDescription extends AbstractBaseControlDescription<'popuplist'> {
+export type PopUpListControlDescription = {
+  control: 'popuplist'
+  label?: string
+  visibleByDefault?: boolean
   defaultValue?: unknown
   options: BasicControlOptions<unknown>
 }
 
-export interface ImportType {
+export type ImportType = {
   source: string // importSource
   name: string | null
   type: 'star' | 'default' | null
```

Comment:
If this is for version bounds, maybe something more evocative? like `versioning`, with descriptors like  'any' | 'latest' | 'default'? Also what's the difference between `null` and `default` ?

Reply:
this is for:

star: import * as Something from
default: import Something from
null: import { Something } from

Reply:
@Rheeseyb I'm actually not sure about this one: `null: import { Something } from`. did i guess it right? :) maybe we could rename it to "named"

Reply:
It is indeed "named". This was changed fairly recently to align with the type used by the parser printer since this needs to be injected into the parsed model, so changing the `type` would also require a change a little later down the line when it comes to trying to use this import.

It's definitely worth doing, but maybe as a second PR immediately after this, because I'm also thinking we should clarify when the `name` field would be `null`, and it would also be better off making separate `ImportType`s to clarify that a bit further. e.g.:
```ts
// import * as name from source
interface ImportStar {
  source: string,
  name: string,
  type: 'star',
}

// import source
interface ImportScript { // Not sure of the best name for this one
  source: string,
  type: 'source',
}
...

type ImportType = ImportStar | ImportNamed | ImportDefault | ImportScript
```